app/scripts/controllers/certificates.js

```diff
@@ -236,4 +247,51 @@ angular.module('openhimWebui2App')
     /****************************************/
 
 
+
+
+    /************************************************/
+    /**         Restart Server Functions           **/
+    /************************************************/
+
+    // server restart later function
+    $scope.restartServerLater = function(){
+      $scope.serverRestartRequired = false;
+    };
+
+    // server restart confirm function
+    $scope.restartServer = function(){
+
+      var restartServer = new Api.Restart();
+      restartServer.$save({}, function(){
+        // restart request sent successfully
+
+        // update restart variables
+        $scope.serverRestarting = true;
+        $scope.serverRestartRequired = false;
+
+        // set estimate time for server restart - 120 seconds
+        $scope.restartTimeout = 10;
```

Comment:
Correct, the server restart is quite quick and doesn't actually take 10seconds. Ive set it to default 10seconds as a kind of buffer. Just thinking in the case of alot of transactions coming through and the server restart is requested. The restart will wait for all transactions being processed to complete first before attempting the server restart. ( No new transactions will come through ). Do you think i should still change this?